src/algebra/char_p/basic.lean

```diff
@@ -475,6 +475,19 @@ end
 
 end
 
+section
```

Comment:
Does this really need its own section?

Reply:
The other sections use more specific type classes than [non_assoc_ring R]. Is there a better solution?

Reply:
See below.